flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/source/reader/RowDataReaderFunction.java

```diff
@@ -61,19 +86,30 @@ public RowDataReaderFunction(
     this.io = io;
     this.encryption = encryption;
     this.filters = filters;
+    this.limit = limit;
   }
 
   @Override
   public DataIterator<RowData> createDataIterator(IcebergSourceSplit split) {
-    return new DataIterator<>(
+    return new LimitableDataIterator<>(
```

Comment:
Is this limit applied after the residual filters?

Reply:
limit is applied after the residual filters. Residual filters are applied inside `RowDataFileScanTaskReader`, which is used and wrapped by the `DataIterator`